net/api/impl/posix.h

```diff
@@ -86,8 +88,8 @@ class HTTPClientPOSIX final {
       // Don't uncomment the next line!
       // connection.SendEOF();
       message_.reset(new HTTPRedirectableReceivedMessage(connection));
-      // TODO(dkorolev): Rename `URL()`, it's only called so now because of HTTP request/response format.
-      const int response_code_as_int = atoi(message_->URL().c_str());
+      // TODO(dkorolev): Rename `Path()`, it's only called so now because of HTTP request/response format.
+      const int response_code_as_int = atoi(message_->Path().c_str());
```

Comment:
It would store that code as ```int```, cast to ```enum class
HTTPResponseCode```, return as ```"Unknown Code"``` when queries as string.
ᐧ

Thanks,
Dima

http://dimakorolev.com/ | http://www.linkedin.com/in/dimakorolev |
dmitry.korolev@gmail.com | +1 (312) 593 2783


On Wed, Jan 7, 2015 at 9:37 AM, John Babak <notifications@github.com> wrote:

> In net/api/impl/posix.h
> <https://github.com/KnowSheet/Bricks/pull/36#discussion-diff-22609661>:
>
> > @@ -86,8 +88,8 @@ class HTTPClientPOSIX final {
> >        // Don't uncomment the next line!
> >        // connection.SendEOF();
> >        message_.reset(new HTTPRedirectableReceivedMessage(connection));
> > -      // TODO(dkorolev): Rename `URL()`, it's only called so now because of HTTP request/response format.
> > -      const int response_code_as_int = atoi(message_->URL().c_str());
> > +      // TODO(dkorolev): Rename `Path()`, it's only called so now because of HTTP request/response format.
> > +      const int response_code_as_int = atoi(message_->Path().c_str());
>
> What would be the behavior of this code in case of a malformed response?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KnowSheet/Bricks/pull/36/files#r22609661>.
>